azkaban-common/src/main/resources/host_name_alias.perl

```diff
@@ -0,0 +1,45 @@
+#!/usr/bin/env perl
+
+# A script to get the VIP name of the current host.
+
+use strict;
+use warnings;
+
+# Get a list of IP addresses that the host has.
+sub get_ip_addrs() {
+    my @ip_addrs = ();
+    # Use the absolute path of ifconfig
+    # since the script may be invoked without PATH being set.
+    for my $line (`/sbin/ifconfig -a`) {
+        # Parse the output of the ifconfig command
+        # looking for patterns of IP addresses
+        if ($line =~ /.*inet addr:(\d+\.\d+\.\d+\.\d+).*/) {
+            push @ip_addrs, $1;
+        }
+    }
+    @ip_addrs;
+}
+
+sub get_hostname_for_ip {
+    my ($ip) = @_;
+    my $result = `getent hosts $ip`;
+    chomp($result);
+    (split(/ /, $result))[-1];
+}
+
+sub get_vip_hostname {
+    my $vip_hostname;
+    for my $ip (get_ip_addrs()) {
+        next if $ip eq '127.0.0.1';
+        $vip_hostname = get_hostname_for_ip($ip);
+        next if $vip_hostname =~ /hcl\d+/;
```

Comment:
> What do you think about adding a config to the config file and allow an external system to set the hostname? Fall back to local host name detection as you we now.

This would be a good feature to have and avoid having to scrape & detect ipv4 vs ipv6 addresses.